phpdotnet/phd/Package/PHP/XHTML.php

```diff
@@ -297,6 +309,30 @@ public function autogenVersionInfo($refnames) {
         $retval = '<p class="verinfo">(' .(htmlspecialchars($verinfo, ENT_QUOTES, "UTF-8")). ')</p>';
         return $retval;
     }
+    private function do_format_type($open, $attrs, $is_return_type) {
+        $retval = '';
+        if ($open) {
+            if (isset($attrs[Reader::XMLNS_DOCBOOK]["class"])) {
+                $this->types = 0;
+            } elseif (isset($this->types)) {
+                if (!$is_return_type && $this->types > 0) $retval .= '|';
+                $this->types++;
+            }
+            if (!$is_return_type) $retval .= '<span class="type">';
+        } else {
+            if (isset($attrs[Reader::XMLNS_DOCBOOK]["class"])) {
+                $this->types = null;
+            }
+            if (!$is_return_type) $retval .= '</span>';
+        }
+        return $retval;
```

Comment:
I find the treatment of return types confusing here. Doesn't this effectively reduce to `if ($is_return_type) return '';` at the top of the function? In which case there's no point for a shared do_format_type method anymore.

Reply:
`do_format_type()` does two things: a) it updates `$num_types` and b) it calculates the `$retval` for `$is_return_type`. Splitting the function into two functions would require to duplicate the code logic; not sure if that would be a simplification.

Reply:
Why do we need num_types for the return types case though?

Reply:
Actually, we don't. Good catch, thanks!